include/Entities/Enemy.hpp

```diff
@@ -2,7 +2,27 @@
 // Created by Ruoying Hao on 10/12/17.
 //
 
-#ifndef ENEMY_HPP
-#define ENEMY_HPP
+#ifndef ENEMY1_HPP
+#define ENEMY1_HPP
+#include "Entities/Entity.hpp"
 
+class Enemy : public Entity{
+public:
+    Enemy();
+    bool isHacked();
+    void setStartPos(sf::Vector2f pos);
+    void setDestPos(sf::Vector2f pos);
+    sf::Vector2f getStartPos();
+    sf::Vector2f getDestPos();
+    //part of path finding, find the next position that the entity should go to
+    sf::Vector2f getNextPos();
```

Comment:
how might this be implemented? I was thinking we store a vector of 'waypoints', and a private int 'current waypoint' (maybe what getNextPos should return) that marks where the enemy is currently trying to reach. They pathfind towards that point. When they reach their current waypoint, the int gets incremented or gets reset to 0 if the last is reached. So we set up the series of waypoints, then the enemy goes from one to the next then returns to the first.

Reply:
This would be something we would need to start implementing. I believe there's more need to be add to the class later.